### PR TITLE
Add possibility to add application-defined configurations for permissions

### PR DESCRIPTION
**Pull request by fraxachun**
Depends on https://github.com/vivid-planet/comet/pull/1654

I don't like the term 'configuration', ideas welcome.

Type-Safety is quite good:
image
(However, it's not perfect: when requiring more than one permission the configurations are merged, the same happens in getPermissionsForUser)

TODO (in next PR): apply this concept to the API-Generator.

---

COM-666

### CODE REVIEW

File: .changeset/wicked-crews-hammer.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Add possibility to add application-defined configurations for permissions
+
+1. Define in API: Add type to Permission-Interface:
+
+```diff
+declare module "@comet/cms-api" {
+    interface Permission {
+-        news: string
++        news: {
++           commentsEdit: boolean;
```
Comment (nsams) [inline .changeset/wicked-crews-hammer.md:15]: I know this is only a made up example, but won't you use a simple string permission "newsCommentsEdit" in pratice?

**Comment by johnnyomair** [comment]
@fraxachun is this PR still ready for review, or do you plan to change it? I can't remember what we discussed in last week's meeting...

File: .changeset/wicked-crews-hammer.md (hunk 1 above)
Comment (johnnyomair) [reply .changeset/wicked-crews-hammer.md:15]: > Nevertheless, the real advantage of this approach is to be able to add custom controls (example: a dropdown which allows you to restrict editing only a certain news-category).

Should we change the example in Demo so it reflects this use case? Or is this overkill?

**Comment by johnnyomair** [comment]
How would this work for settings permissions on entity level? I believe that will be the main use case for permission configuration. Shouldn't the configuration then store an array of allowed IDs?

**Pull request closed without merging**